airflow/hooks/http_hook.py

```diff
@@ -85,34 +101,36 @@ def run(self, endpoint, data=None, headers=None, extra_options=None):
         self.log.info("Sending '%s' to url: %s", self.method, url)
         return self.run_and_check(session, prepped_request, extra_options)
 
+    def check_response(self, response):
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
+            self.log.error("HTTP error: %s", response.reason)
+            if self.method not in ['GET', 'HEAD']:
+                self.log.error(response.text)
+            raise AirflowException(str(response.status_code) + ":" + response.reason)
+
     def run_and_check(self, session, prepped_request, extra_options):
         """
         Grabs extra options like timeout and actually runs the request,
         checking for the result
         """
         extra_options = extra_options or {}
 
-        response = session.send(
-            prepped_request,
-            stream=extra_options.get("stream", False),
-            verify=extra_options.get("verify", False),
-            proxies=extra_options.get("proxies", {}),
-            cert=extra_options.get("cert"),
-            timeout=extra_options.get("timeout"),
-            allow_redirects=extra_options.get("allow_redirects", True))
-
         try:
-            response.raise_for_status()
-        except requests.exceptions.HTTPError:
-            # Tried rewrapping, but not supported. This way, it's possible
-            # to get reason and code for failure by checking first 3 chars
-            # for the code, or do a split on ':'
-            self.log.error("HTTP error: %s", response.reason)
-            if self.method not in ('GET', 'HEAD'):
-                # The sensor uses GET, so this prevents filling up the log
-                # with the body every time the GET 'misses'.
-                # That's ok to do, because GETs should be repeatable and
-                # all data should be visible in the log (no post data)
-                self.log.error(response.text)
-            raise AirflowException(str(response.status_code)+":"+response.reason)
-        return response
+            response = session.send(
+                prepped_request,
+                stream=extra_options.get("stream", False),
+                verify=extra_options.get("verify", False),
+                proxies=extra_options.get("proxies", {}),
+                cert=extra_options.get("cert"),
+                timeout=extra_options.get("timeout"),
+                allow_redirects=extra_options.get("allow_redirects", True))
+
+            if extra_options.get('check_response', True):
+                self.check_response(response)
+            return response
+
+        except requests.exceptions.ConnectionError as ex:
+            logging.error(str(ex) + 'add retry logic here')
```

Comment:
Oh missed this first time -this should be `self.log.error` ... and we probably want something else here?